compiler/modules/front/const.bal

```diff
@@ -51,7 +51,9 @@ function resolveConstDefn(ModuleTable mod, s:ConstDefn defn) returns s:ResolvedC
     else {
         defn.resolved = false;
         ConstFoldContext cx = new ConstFoldContext(defn.name, mod);
-        s:Expr expr = check foldExpr(cx, (), defn.expr);
+        s:InlineBasicTypeDesc? td = defn.td;
+        t:SemType? expectedType = td == () ? () : s:resolveInlineTypeDesc(td);
+        s:Expr expr = check foldExpr(cx, expectedType, defn.expr);
```

Comment:
You also need to check that the value does belong to the type. For example, this is illegal

```
const boolean b = 17;
```